lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -1,45 +1,69 @@
 
-function restrictPropCoreFunctions()
-    local disallowedRanks = {}
-    disallowedRanks["user"] = true
-    disallowedRanks["regular"] = true
-
-    local restrictedFunctions = {
-        "propSpawn(sn)",
-        "propSpawn(en)",
-        "propSpawn(svn)",
-        "propSpawn(evn)",
-        "propSpawn(san)",
-        "propSpawn(ean)",
-        "propSpawn(svan)",
-        "propSpawn(evan)",
-        "seatSpawn(sn)",
-        "seatSpawn(svan)",
-        "setPos(e:v)",
-        "reposition(e:v)",
-        "propManipulate(e:vannn)"
-        --"propBreak(e:)"
-    }
-
-    for _, signature in pairs( restrictedFunctions ) do
-        if wire_expression2_funcs then
-            local oldFunc = wire_expression2_funcs[signature][3]
-
-            wire_expression2_funcs[signature][3] = function( self, ... )
-                if disallowedRanks[self.player:GetUserGroup()] == nil then
-                    local isInBuildMode = self.player:GetNWBool("CFC_PvP_Mode", false) == false
-
-                    if isInBuildMode or self.player:IsAdmin() then
-                        return oldFunc( self, ... )
-                    else
-                        self.player:ChatPrint( "You can't use PropCore in PvP mode" )
-                    end
-                else
-                    self.player:ChatPrint( "You don't have access to " .. signature )
-                end
+local disallowedRanks = {}
+disallowedRanks["user"] = true
+disallowedRanks["regular"] = true
+
+local restrictedFunctions = {
+    "propSpawn(sn)",
+    "propSpawn(en)",
+    "propSpawn(svn)",
+    "propSpawn(evn)",
+    "propSpawn(san)",
+    "propSpawn(ean)",
+    "propSpawn(svan)",
+    "propSpawn(evan)",
+    "seatSpawn(sn)",
+    "seatSpawn(svan)",
+    "setPos(e:v)",
+    "reposition(e:v)",
+    "propManipulate(e:vannn)"
+    --"propBreak(e:)"
+}
+
+local adminOnlyFunctions = {
+    "use(e:)"
+}
+local function adminOnlyCondition( self, ... )
+    if self.player:IsAdmin() then
+        return true
+    end
+
+    return false, "Only Admins can use this function"
+end
+
+local function restrictedCondition( self, ... )
+    local isInBuildMode = self.player:GetNWBool( "CFC_PvP_Mode" ) == false
+
+    if disallowedRanks[self.player:GetUserGroup()] then
+        return false, "You don't have access to this function"
+    end
+
+
+    if not isInBuildMode and not self.player:IsAdmin() then
+        return false, "You can't use propcore in PvP"
+    end
+
+    return true
+end
+
+local function restrict( signatures, condition )
+    for _, signature in pairs( signatures ) do
+        local oldFunc = wire_expression2_funcs[signature][3]
+
+        wire_expression2_funcs[signature][3] = function( self, ... )
+            local canRun, reason = condition( self, ... )
+
+            if canRun then
+                return oldFunc( self, ... )
+            else
+                self.player:ChatPrint( "Couldn't run " .. signature .. ":" .. reason )
             end
```

Comment:
Let's change the default behavior and reduce some conditionals here:

```suggestion
            if not canRun then
                return self.player:ChatPrint( "Couldn't run " .. signature .. ":" .. reason )
            end
            
            return oldFunc( self, ... )
```